main/src/main/scala/sbt/internal/Load.scala

```diff
@@ -988,19 +1045,25 @@ private[sbt] object Load {
       }
   }
 
+  /** Variable to control the indentation of the timing logs. */
+  private var timedIndentation: Int = 0
+
   /** Debugging method to time how long it takes to run various compilation tasks. */
   private[sbt] def timed[T](label: String, log: Logger)(t: => T): T = {
+    timedIndentation += 1
     val start = System.nanoTime
     val result = t
     val elapsed = System.nanoTime - start
-    log.debug(label + " took " + (elapsed / 1e6) + " ms")
+    timedIndentation -= 1
```

Comment:
Thanks for having a look at this @dwijnand. These are the reasons why I discarded those ideas:
* The execution of the sbt load is sequential and `timed` is only used within the `Load` object.
* If an exception is thrown, sbt stops, reports it and exits. Given these circumstances, there should be no corruption of the value of indentation.
  
What we could do, however, is to make this utility global and accessible to other sbt subsystems where these conditions don't hold. But I haven't had the need to compute the time of other sbt parts, so that's why I didn't consider it in the first place.